### DISCUSSION TITLE
Reporting sea level pressure from Lat & Long

### DISCUSSION THREAD

**Discussion by robertrau** [weather-gov/api#771] [Q&A]
I need to get sea level pressure but I am starting with latitude and longitude location data. So far the only thing I have come up with is:
Using
api.weather.gov/points/{latitude},{longitude}/
and then parsing out the forcast office ID and grid points,
then using
api.weather.gov/gridpoints/{wfo}/{x},{y}/stations
to give me a station, and finally using
api.weather.gov/stations/{station}/observations/latest
to get sea level pressure.

I'm sure there is a better way.

I am grateful for any help.

**Comment by dwhitemv25** [comment]
You have the right order of operations. The `observationStations` property of the `/points` query gives you the `/gridpoints/.../stations` link so you don't have to build your own gridpoints URL. You can also cache the results of the earlier queries as they change very infrequently.

Be aware that not all stations report SLP or report it at every interval. ASOS stations (usually large airports) are your best bet for consistent SLP observations. Otherwise you may have to convert from station barometric pressure or altimeter setting, depending on what's available.

**Reply by robertrau** [reply]
Thank you. I see, several stations don't supply SLP. For converting from "altimeter setting" I also need current surface barametric pressure, don't I?
Can you point me to a link for this conversion?
Thank you for the help.

**Reply by dwhitemv25** [reply]
Reversing altimeter to station pressure is simple, but calculating sea level pressure is complicated.  At a minimum you will need the station altitude. Ideally you will also want temperature and humidity (and/or dewpoint; with 2 of the values you can calculate the 3rd with psychometric formulas). 

General meteorologic formulas can be found at:
- https://www.weather.gov/epz/wxcalc

A formula for getting station pressure from altimeter is here:
- https://www.weather.gov/media/epz/wxcalc/stationPressure.pdf

Converting station pressure to sea level pressure is more complex, depending on the level of accuracy you want. There are corrections for temperature and humidity (and air density, gravity, and the aspheric nature of the Earth if you want to get crazy), and some of the constants are from precomputed tables. The best sources I can find for this are:

- https://en.wikipedia.org/wiki/Barometric_formula
- https://integritext.net/DrKFS/correctiontosealevel.htm
- [Davis technote on derived values in their weather station products](https://cdn.shopify.com/s/files/1/0515/5992/3873/files/App_Note_28_Rev_G_Derived_Variables.pdf?v=1718824378)

You may just have to choose or develop your own adjustment so your data is internally consistent.

**Comment by robertrau** [comment]
I am sometimes getting barometric pressure readings from KDET that aren't making sense to me. Here is part of the JSON response @ about 11:30 AM 7/22/2025:
    "properties": {
        "@id": "https://api.weather.gov/stations/KDET/observations/2025-07-22T13:53:00+00:00",
        "@type": "wx:ObservationStation",
        "elevation": {
            "unitCode": "wmoUnit:m",
            "value": 191
        },
        "station": "https://api.weather.gov/stations/KDET",
        "stationId": "KDET",
        "stationName": "Coleman A Young Municipal Airport",
        "timestamp": "2025-07-22T13:53:00+00:00",
        "rawMessage": "KDET 221353Z 03003KT 10SM CLR 20/13 A3016 RMK AO2 SLP211 T02000133",
        "textDescription": "Clear",
        "icon": "https://api.weather.gov/icons/land/day/skc?size=medium",
        "presentWeather": [],
        "temperature": {
            "unitCode": "wmoUnit:degC",
            "value": 20,
            "qualityControl": "V"
        },
        "dewpoint": {
            "unitCode": "wmoUnit:degC",
            "value": 13.3,
            "qualityControl": "V"
        },
        "windDirection": {
            "unitCode": "wmoUnit:degree_(angle)",
            "value": 30,
            "qualityControl": "V"
        },
        "windSpeed": {
            "unitCode": "wmoUnit:km_h-1",
            "value": 5.4,
            "qualityControl": "V"
        },
        "windGust": {
            "unitCode": "wmoUnit:km_h-1",
            "value": null,
            "qualityControl": "Z"
        },
        "barometricPressure": {
            "unitCode": "wmoUnit:Pa",
            "value": 102130,
            "qualityControl": "V"
        },
        "seaLevelPressure": {
            "unitCode": "wmoUnit:Pa",
            "value": 102110,
            "qualityControl": "V"
        },

By my math, for a sea level presssure of 102110 Pa, and a station altitude of 191 meters, the surface barometric measurement would have to be around 99800, not the 102130 reported. And it received a quality of "V".
What am I missing here?

Thank you in advance.

**Reply by dwhitemv25** [reply]
For that report, 1021.30 would be the altimeter setting, not station pressure.

The raw METAR:
```
KDET 221353Z 03003KT 10SM CLR 20/13 A3016 RMK AO2 SLP211 T02000133
```

'A' is the altimeter and was reported at 30.16 inHg. Sea level pressure is 'SLP' and is the last 3 digits in mb, so SLP211 becomes 1021.1 mb. 

ASOS METAR reporting stations (airports, predominantly) send altimiter on every report and sea level pressure hourly.

**Reply by rr0ss0rr** [reply]
Can you explain `SLP211 becomes 1021.1 mb` where does the `10` prefix comes from?

**Comment by robertrau** [comment]
I am going to guess that the only possible pressures are above 950mb and below 1050mb (or similar thresholds) so for values lower than 500 you prefix with 10 otherwise you prefix with 9. The METAR format goes back to the later 1960s when data rates were slow, so every character mattered for speed.
This is only an educated guess.

**Reply by rr0ss0rr** [reply]
Thanks
